scripts/findhaikus.py

```diff
@@ -34,7 +34,7 @@
     text = open(sys.argv[1], "r").read()
 except:
     text = sys.stdin.read()
-for haiku in HaikuFinder(text.decode('utf-8')).find_haikus():
+for haiku in HaikuFinder(unicode(text, errors='replace')).find_haikus():
```

Comment:
This is incorrect, as it treats the input as ASCII, and throws away many legitimate utf-8 encoded characters. Try```python
unicode(text, encoding='utf-8', errors='replace')
```